contracts/EuroToken.sol

```diff
@@ -0,0 +1,235 @@
+    /// @dev proof is provided in form of log entry on which EURT_DEPOSIT_MANAGER
+    /// @dev will act off-chain to return required Euro amount to EUR-T holder
+    function withdraw(uint256 amount)
+        public
+    {
+        require(_balances[msg.sender] >= amount);
+        _balances[msg.sender] -= amount;
+        _totalSupply -= amount;
+        Withdrawal(msg.sender, amount);
+    }
+
+    /// @notice enables or disables address to be receipient of EUR-T
+    function setAllowedTransferTo(address to, bool allowed)
+        public
+        only(ROLE_EURT_DEPOSIT_MANAGER)
+    {
+        _allowedTransferTo[to] = allowed;
+        AllowedToAddress(to, allowed);
+    }
+
+    /// @notice enables or disables address to be sender of EUR-T
+    function setAllowedTransferFrom(address from, bool allowed)
+        public
+        only(ROLE_EURT_DEPOSIT_MANAGER)
+    {
+        _allowedTransferFrom[from] = allowed;
+        AllowedFromAddress(from, allowed);
+    }
+
+    function allowedTransferTo(address to)
+        public
+        constant
+        returns (bool)
+    {
+        return _allowedTransferTo[to];
+    }
+
+    function allowedTransferFrom(address from)
+        public
+        constant
+        returns (bool)
+    {
+        return _allowedTransferFrom[from];
+    }
+
+    //
+    // Overrides ERC20 Interface to allow transfer from/to allowed addresses
+    //
+
+    function transfer(address to, uint256 amount)
+        onlyAllowedTransferFrom(msg.sender)
+        onlyAllowedTransferTo(to)
+        public
+        returns (bool success)
+    {
+        return BasicToken.transfer(to, amount);
+    }
+
+    function transferFrom(address from, address to, uint256 amount)
+        onlyAllowedTransferFrom(from)
+        onlyAllowedTransferTo(to)
+        public
+        returns (bool success)
+    {
+        return StandardToken.transferFrom(from, to, amount);
+    }
+
+    //
+    // Overrides migration source
+    //
+
+    function migrate()
+        public
+        onlyMigrationEnabled
+        onlyAllowedTransferTo(msg.sender)
+    {
+        // burn deposit
+        uint256 amount = _balances[msg.sender];
+        require( amount > 0);
+        _balances[msg.sender] -= amount;
+        _totalSupply -= amount;
+        // migrate to
+        bool success = EuroTokenMigrationTarget(_migration).migrateOwner(msg.sender, amount);
+        require(success);
+        // set event
+        OwnerMigrated(msg.sender, amount);
+    }
+
+    //
+    // Implements IERC677Token
+    //
+
+    function approveAndCall(address spender, uint256 amount, bytes extraData)
+        returns (bool success)
+    {
+        require(approve(spender, amount));
+
+        success = IERC677Callback(spender).receiveApproval(
+            msg.sender,
+            amount,
+            this,
+            extraData
+        );
+
+        return success;
+    }
+
+    //
+    // Overrides Reclaimable
```

Comment:
This whole part can be left out. It is not necessary to override if nothing is changed from the default behaviour.

Reply:
true, I will comment it out but leave the note